gen/sam/group_generator.py

```diff
@@ -8,16 +8,20 @@
 
 class GroupGenerator():
 
-    def __init__(self, game, ground_object):
+    def __init__(self, game, ground_object, faction = None): # faction is not mandatory because some subclasses do not use it
```

Comment:
But why don't they use it? What's the value in not requiring this? If not all subclasses use it, why is it part of the base class's constructor?

Reply:
You're right, I could have done it in another, better way.
Man do I learn a lot with you ! 
I'll push a change.